src/sync/sync_session.cpp

```diff
@@ -473,18 +474,22 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
     }
 }
 
-void SyncSession::NotifierPackage::update(const Progress& current_progress)
+void SyncSession::NotifierPackage::update(const Progress& current_progress, bool data_is_fresh)
 {
-    if (is_streaming || captured_transferrable)
+    if (is_streaming || captured_transferrable || !data_is_fresh)
         return;
 
     captured_transferrable = direction == NotifierType::download ? current_progress.downloadable
                                                                  : current_progress.uploadable;
 }
 
-std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress, bool& is_expired) const
+std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress,
+                                                                      bool& is_expired) const
 {
-    REALM_ASSERT(is_streaming || captured_transferrable);
+    // It's possible for a non-streaming notifier to not yet have fresh transferrable bytes data.
+    // In that case, we don't call it at all.
```

Comment:
It's probably a result of my reading only the diff rather than looking at the surrounding context (in this case, the caller of the method). I'd incorrectly thought this was only called once when registering the notification, rather than each time new progress information is available. My mistake.